lib/compiler/opcodes.js

```diff
@@ -4,52 +4,52 @@
 const opcodes = {
   // Stack Manipulation
 
-  PUSH:             0,    // PUSH c
```

Comment:
Doesn't this (as an example of several opcode changes) [conflict](https://github.com/metadevpro/ts-pegjs/blob/f2d5caca50d5c833f643a608bf6959c8c9a7b9da/src/passes/generate-ts.js#L247) with the typescript plugin?  Is  @metadevpro following the work here yet?  How could they support both peggy and pegjs if they wanted?

Reply:
It will. Instead of renaming that opcode we can just leave it unused and introduce a new opcode for `PUSH_EMPTY_STRING`. Renaming `MATCH_REGEXP` also should be reverted. Also we could add an alias instead of renaming or just do nothing. Then in version 2.0 we could completely remove the unused opcode.

Reply:
OK, my suggestion from your choices above:

- Leave `PUSH` as 0.  Mark it as deprecated with a comment.
- Create a new opcode for `PUSH_EMPTY_STRING`
- Leave `MATCH_CHAR_CLASS` and `MATCH_REGEXP` both as opcode 20, but mark `MATCH_REGEXP`.  Mark `MATCH_REGEXP` as deprecated with a comment.

I think this will allow plugins to support both peggy and pegjs at the same time adequately.  If you expect that plugins would need to take different actions for `MATCH_CHAR_CLASS` and `MATCH_REGEXP`, then don't alias them together, and instead leave `MATCH_REGEXP` as opcode 20, create a new opcode for `MATCH_CHAR_CLASS`.

Reply:
All done as you said. I've make a rebase , changed only the last and the pre-pre-last commits. `MATCH_REGEXP` and `PUSH` marked `@deprecated`, noted this in the `CHANGELOG.md`

Reply:
Thanks!